src/math/mat4.h

```diff
@@ -202,19 +378,29 @@ namespace Bald::Math {
     constexpr Mat4 Mat4::Multiply(const Mat4& other) const noexcept {
         Mat4 result;
 
-        for (int col = 0; col < 4; ++col)
-            for (int row = 0; row < 4; ++row)
-                result.m_MatrixElements[row + col * 4] = m_MatrixElements[4 * col] * other.m_MatrixElements[row] +
-                                                         m_MatrixElements[4 * col + 1] *
-                                                         other.m_MatrixElements[row + 1 * 4] +
-                                                         m_MatrixElements[4 * col + 2] *
-                                                         other.m_MatrixElements[row + 2 * 4] +
-                                                         m_MatrixElements[4 * col + 3] *
-                                                         other.m_MatrixElements[row + 3 * 4];
+        for (int row = 0; row < 4; ++row)
+            for (int col = 0; col < 4; ++col)
+                result.m_MatrixElements[col + row * 4] = m_MatrixElements[col]      * other.m_MatrixElements[4 * row]     +
+                                                         m_MatrixElements[col + 4]  * other.m_MatrixElements[1 + row * 4] +
+                                                         m_MatrixElements[col + 8]  * other.m_MatrixElements[2 + row * 4] +
+                                                         m_MatrixElements[col + 12] * other.m_MatrixElements[3 + row * 4];
 
         return result;
     }
 
+    constexpr Vec4 Mat4::Multiply(const Vec4& other) const noexcept {
+        float x = m_MatrixElements[0] * other.GetX() + m_MatrixElements[0 + 1 * 4] * other.GetY() + m_MatrixElements[0 + 2 * 4] * other.GetZ() + m_MatrixElements[0 + 3 * 4] * other.GetW();
+        float y = m_MatrixElements[1] * other.GetX() + m_MatrixElements[1 + 1 * 4] * other.GetY() + m_MatrixElements[1 + 2 * 4] * other.GetZ() + m_MatrixElements[1 + 3 * 4] * other.GetW();
+        float z = m_MatrixElements[2] * other.GetX() + m_MatrixElements[2 + 1 * 4] * other.GetY() + m_MatrixElements[2 + 2 * 4] * other.GetZ() + m_MatrixElements[2 + 3 * 4] * other.GetW();
+        float w = m_MatrixElements[3] * other.GetX() + m_MatrixElements[3 + 1 * 4] * other.GetY() + m_MatrixElements[3 + 2 * 4] * other.GetZ() + m_MatrixElements[3 + 3 * 4] * other.GetW();
+
```

Comment:
Is this worth:
float tempX = other.GetX()
float tempY = other.GetY()

Reply:
It will most likely be optimized by a compiler, we can ask Grzesiek what he thinks of it.

Reply:
IMO I would do that. Additionally we can think about one function which returns all coords of certain vector.

Reply:
Corrected :+1: